src/FTGlyph/FTPixmapGlyph.cpp

```diff
@@ -40,18 +41,30 @@
 
 
 FTPixmapGlyph::FTPixmapGlyph(FT_GlyphSlot glyph) :
-    FTGlyph(new FTPixmapGlyphImpl(glyph))
+    FTGlyph(NewImpl(glyph))
 {}
 
 
 FTPixmapGlyph::~FTPixmapGlyph()
 {}
 
 
+FTGlyphImpl *FTPixmapGlyph::NewImpl(FT_GlyphSlot glyph)
+{
+  FTPixmapGlyphImpl *Impl = new FTPixmapGlyphImpl(glyph);
+  if (Impl->destWidth && Impl->destHeight)
+    return Impl;
+  return new FTBitmapGlyphImpl(glyph);
```

Comment:
There is a memory leak here (Impl).

Using unique pointers (http://en.cppreference.com/w/cpp/memory/unique_ptr) would allow to handle that in a nice way.